e2e/test/TwinE2ETests.cs

```diff
@@ -202,52 +202,50 @@ public async Task Twin_ClientSetsReportedPropertyWithoutDesiredPropertyCallback(
         {
             // arrange
 
-            TestDevice testDevice = await TestDevice.GetTestDeviceAsync(DevicePrefix).ConfigureAwait(false);
-            using (var deviceClient = DeviceClient.CreateFromConnectionString(testDevice.ConnectionString, transportType))
-            {
-                await Twin_DeviceSetsReportedPropertyAndGetsItBack(deviceClient).ConfigureAwait(false);
+            using TestDevice testDevice = await TestDevice.GetTestDeviceAsync(_devicePrefix).ConfigureAwait(false);
+            using var deviceClient = DeviceClient.CreateFromConnectionString(testDevice.ConnectionString, transportType);
 
-                int connectionStatusChangeCount = 0;
-                ConnectionStatusChangesHandler connectionStatusChangesHandler = (ConnectionStatus status, ConnectionStatusChangeReason reason) =>
-                {
-                    Interlocked.Increment(ref connectionStatusChangeCount);
-                };
+            await Twin_DeviceSetsReportedPropertyAndGetsItBack(deviceClient).ConfigureAwait(false);
 
-                string propName = Guid.NewGuid().ToString();
-                string propValue = Guid.NewGuid().ToString();
+            int connectionStatusChangeCount = 0;
+            ConnectionStatusChangesHandler connectionStatusChangesHandler = (ConnectionStatus status, ConnectionStatusChangeReason reason) =>
+            {
+                Interlocked.Increment(ref connectionStatusChangeCount);
+            };
 
-                _log.WriteLine($"{nameof(Twin_ServiceSetsDesiredPropertyAndDeviceReceivesEvent)}: name={propName}, value={propValue}");
+            string propName = Guid.NewGuid().ToString();
+            string propValue = Guid.NewGuid().ToString();
 
-                // act
-                await RegistryManagerUpdateDesiredPropertyAsync(testDevice.Id, propName, propValue).ConfigureAwait(false);
-                await Task.Delay(TimeSpan.FromSeconds(10)).ConfigureAwait(false);
+            s_log.WriteLine($"{nameof(Twin_ServiceSetsDesiredPropertyAndDeviceReceivesEventAsync)}: name={propName}, value={propValue}");
 
-                // assert
-                Assert.AreEqual(0, connectionStatusChangeCount, "AMQP should not be disconnected.");
+            // act
+            await RegistryManagerUpdateDesiredPropertyAsync(testDevice.Id, propName, propValue).ConfigureAwait(false);
+            await Task.Delay(TimeSpan.FromSeconds(10)).ConfigureAwait(false);
 
-                // clean up
-                await deviceClient.CloseAsync().ConfigureAwait(false);
-            }
+            // assert
+            Assert.AreEqual(0, connectionStatusChangeCount, "AMQP should not be disconnected.");
+
+            // clean up
+            await deviceClient.CloseAsync().ConfigureAwait(false);
```

Comment:
does the using statement already close the device client for us?

Reply:
The end effect of disposing the device client instance would be that transport layer is marked as closed, retry policies/ token refreshers etc are all cancelled gracefully, and then the object is marked as disposed. So yes, disposing the client would essentially close the connection and mark the resources as disposed.